blocks/recentreports/recentreports.js

```diff
@@ -0,0 +1,253 @@
+  carouselBlockParent.classList.add('carousel-wrapper');
+  carouselBlockParent.appendChild(carouselBlock);
+  block.insertBefore(carouselBlockParent, block.firstChild.nextSibling);
+  decorateBlock(carouselBlock);
+
+  loadBlock(carouselBlock).then((recentreportblock) => {
+    let totalSlidesDisplayed = 0;
+    if (carouselItems.children.length >= carouselBlock.dataset.visibleSlides) {
+      totalSlidesDisplayed = carouselBlock.dataset.visibleSlides;
+      carouselBlock.style.setProperty('--displayed-slides', carouselBlock.dataset.visibleSlides);
+    } else {
+      totalSlidesDisplayed = carouselItems.children.length;
+      carouselBlock.style.setProperty('--displayed-slides', carouselItems.children.length);
+    }
+
+    if (totalSlidesDisplayed > 1) {
+      let maxWidth = 0;
+      carouselBlock.querySelectorAll('.carousel-slide .box').forEach((slide) => {
+        if (maxWidth < slide.offsetWidth) {
+          maxWidth = slide.offsetWidth;
+        }
+      });
+
+      const margin = (carouselBlock.parentElement.offsetWidth
+        - (totalSlidesDisplayed * maxWidth)) / 2;
+      carouselBlock.querySelector('.carousel-slides-container').style
+        .setProperty('margin-inline', `${margin.toString()}px`);
+    }
+    recentreportblock.style.visibility = 'visible';
+  });
+}
+
+function handleTitleConfig(titleElement, container) {
+  const titleText = titleElement.textContent.trim();
+  const title = document.createElement('h2');
+  title.textContent = titleText;
+
+  const titleWrapper = document.createElement('div');
+  titleWrapper.classList.add('title-wrapper');
+  titleWrapper.appendChild(title);
+
+  container.insertBefore(titleWrapper, container.firstChild);
+}
+
+function addDiscoverLink(discoverMoreDiv, block) {
+  const discoverMoreAnchor = discoverMoreDiv.querySelector('a');
+  if (discoverMoreAnchor) {
+    const div = document.createElement('div');
+    div.className = 'discover-more-link';
+    const anchor = document.createElement('a');
+    anchor.href = discoverMoreAnchor.href;
+    anchor.className = 'link-color';
+    anchor.target = '_blank';
+    anchor.textContent = discoverMoreAnchor.title;
+    const icon = document.createElement('i');
+    icon.className = 'icon-up-arrow icon ';
+    anchor.appendChild(icon);
+    div.appendChild(anchor);
+    block.appendChild(div);
+  }
+}
+
+export default async function decorate(block) {
+  const configElementsArray = Array.from(block.children);
+  configElementsArray.map(async (configElement) => {
+    configElement.style.display = 'none';
+    const configNameElement = configElement.querySelector('div');
+    const configName = configNameElement.textContent.trim().toLowerCase();
+    const blockCfg = readBlockConfig(block);
+    if (configName === 'type') {
+      const carouselItems = document.createElement('div');
+      carouselItems.classList.add('carousel-items');
+      const apiName = configNameElement.nextElementSibling.textContent.trim();
```

Comment:
The author user may not understand the API type, it is not authorable in other blocks, so it would be great to keep the same.

Reply:
Have changed the logic so now I am not taking api name from the author but just a business key